internal/configs/configschema/coerce_value.go

```diff
@@ -71,6 +71,11 @@ func (b *Block) coerceValue(in cty.Value, path cty.Path) (cty.Value, error) {
 			return cty.UnknownVal(impliedType), path.NewErrorf("attribute %q is required", name)
 		}
 
+		// This might break backwards compatability
+		if !attrS.Computed && !attrS.Optional && val.IsNull() {
```

Comment:
**(question)** What if a required attribute is allowed to be `nil`/`null`? 

IMO there are two things separate validations:
- Attribute is provided, or not
- Provided attribute is nil/null, or not

I feel that nilabilitiy validation shall be at the level of backend configuration instead.

**(nitpick)** AFAIK `!attrS.Computed && !attrS.Optional` is equivalent to `attrS.Required`. The taler is easier to understand when reading IMO.

Reply:
Unfortunately, due to the confusion of Optional vs Required, we can't rely on Required to be set correctly.  It's a very tricky situation.

Reply:
Part of the problem is that the backend configuration sets all fields to null by default, so this bypasses validation for the most common cases.  Since that is not easy to fix, I'm curious if there is a real use case for allowing null for a required field.  To me, that seems invalid.

Reply:
The proposal when we shall not expect users setting attributes to `null` makes absolute sense. An illustration:
```terraform
terraform {
  cloud {
    organization = "<MY_CLOUD_ORGANIZATION>"
    hostname     = "<MY_CLOUD_BACKEND_HOSTNAME>"

    workspaces {
      name = "foo"
      tags = null
    }
  }
}
```
will result in the error `Invalid workspaces configuration: Only one of workspace "tags" or "name" is allowed.`.

In other words, this thread can be resolved :) Thank you!